brian2genn/templates/max_row_length_array.cpp

```diff
@@ -0,0 +1,58 @@
+{# USES_VARIABLES { _synaptic_pre, _synaptic_post, sources, targets,
+                    N_incoming, N_outgoing, N,
+                    N_pre, N_post, _source_offset, _target_offset }
+#}
+{# WRITES_TO_READ_ONLY_VARIABLES { N_incoming, N_outgoing}
+#}
+
+#include "brianlib/common_math.h"
+#include "brianlib/stdint_compat.h"
+#include<cmath>
+#include<ctime>
+#include<iostream>
+#include<fstream>
+#include<climits>
+#include "brianlib/stdint_compat.h"
+#include "synapses_classes.h"
+
+////// SUPPORT CODE ///////
+namespace {{codeobj_name}}_array {
+	{{support_code_lines|autoindent}}
+}
+
+////// HASH DEFINES ///////
+{{hashdefine_lines|autoindent}}
+
+void _run_{{codeobj_name}}()
+{
+    using namespace brian;
+    using namespace {{codeobj_name}}_array;
+
+    ///// CONSTANTS ///////////
+    %CONSTANTS%
+
+    ///// POINTERS ////////////
+    {{pointers_lines|autoindent}}
+  
+    const size_t _old_num_synapses = {{N}};
+    const size_t _new_num_synapses = _old_num_synapses + _numsources;
+
+    {# Get N_post and N_pre in the correct way, regardless of whether they are
+	constants or scalar arrays#}
+    const size_t _N_pre = {{constant_or_scalar('N_pre', variables['N_pre'])}};
+    const size_t _N_post = {{constant_or_scalar('N_post', variables['N_post'])}};
+    {{_dynamic_N_incoming}}.resize(_N_post + _target_offset);
+    {{_dynamic_N_outgoing}}.resize(_N_pre + _source_offset);
+
+    for (size_t _idx=0; _idx<_numsources; _idx++) {
+      {# After this code has been executed, the arrays _real_sources and
+	  _real_variables contain the final indices. Having any code here it all is
```

Comment:
I'm guessing ``_real_variables`` should be ``_real_targets``

Reply:
I agree